.github/workflows/pr-checks.yml

```diff
@@ -0,0 +1,41 @@
+name: PR Guideline Checks
+
+on:
+  pull_request:
+    types: [opened, reopened, edited]
+
+jobs:
+  pr-checks:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: install PR check dependencies
+        run: |
+          pip install -r ${{ github.workspace }}/.github/workflows/requirements.txt
+
+      - name: Run PR checks
+        id: pr_checks
+        continue-on-error: true
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          PR_ID=$(echo $GITHUB_REF | cut -d/ -f3)
+          cd ${{ github.workspace }}
+          .github/workflows/pr-check ${{ github.repository }} $PR_ID > pr_checks.txt
+
+      - name: Post comment if checks didn't pass
+        uses: actions/github-script@v5
+        if: steps.pr_checks.outcome == 'failure'
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
```

Comment:
surely best if you can do this as part of the main action - you already have the `PyGithub` dependency and `secrets.GITHUB_TOKEN`